opencog/cogserver/server/Request.cc

```diff
@@ -45,14 +47,41 @@ Request::~Request()
         _console->OnRequestComplete();
         _console->put();  // dec use count we are done with it.
     }
+
+    // Normally the thread object will be cleared when it is done.
+    if (_thread)
+    {
+        if (_thread->joinable())
+            _thread->join();
+        delete _thread;
+    }
+}
+
+void Request::executeAsynch(void)
+{
+    // Execute in a new thread.
+    auto execute_wrapper = [&](void) 
+    {
+        try{
+            execute();
+        } catch (...) {
+            logger().debug("Request::executeAsynch caught exception!");
+        }        
+        
+        // Always tell the CogServer so this request is deleted
+        // even in the event of an exception.
+        _cogserver.asynchRequestDone(this);
+    };
+    
+    _thread = new std::thread(execute_wrapper);
```

Comment:
Eh? Perhaps I mis-spoke or ms-lead. The built-in requests cannot be run async, because they are not thread-safe. And because they are a stunningly boring collection, doing amazingly boring and trite things. So running those async is pointless, when not outright wrong. 

That leaves the scheme and the python shells. These shells run commands in sequential order, when you are in one shell, and that makes sense because the user expects sequential ordering. 

However, if you open separate telnet connections, these run independently and async  to each other -- they get different threads from a thread-pool of evaluators that are hot, primed and ready to go.  That is, the shells are already asynch exactly where they need to be, and exactly synch where they need to be.

So .. since requests should never be asynch, this new complexity does nothing except to increase complexity. It serves no practical purpose.